SMLHelper/Handlers/OptionsPanelHandler.cs

```diff
@@ -32,11 +35,45 @@ public static void RegisterModOptions(ModOptions options)
         /// <summary>
         /// Registers your mod options to the in-game menu.
         /// </summary>
-        /// <param name="options">The mod options. Create a new child class inheriting from this one and add your options to it.</param>
+        /// <param name="options">The mod options. Create a new child class inheriting from this one
+        /// and add your options to it.</param>
         /// <seealso cref="ModOptions"/>
         void IOptionsPanelHandler.RegisterModOptions(ModOptions options)
         {
             OptionsPanelPatcher.modOptions.Add(options.Name, options);
         }
+
+        /// <summary>
+        /// Generates an options menu based on the attributes and members declared in the <see cref="ConfigFile"/>
+        /// and registers it to the in-game menu.
+        /// </summary>
+        /// <typeparam name="T">A class derived from <see cref="ConfigFile"/> to generate the options menu from.</typeparam>
+        /// <returns>An instance of the <typeparamref name="T"/> : <see cref="ConfigFile"/> with values loaded
+        /// from the config file on disk.</returns>
+        public static T RegisterModOptions<T>() where T : ConfigFile, new()
+            => Main.RegisterModOptions<T>();
+
+        /// <summary>
+        /// Generates an options menu based on the attributes and members declared in the <see cref="ConfigFile"/>
+        /// and registers it to the in-game menu.
+        /// </summary>
+        /// <typeparam name="T">A class derived from <see cref="ConfigFile"/> to generate the options menu from.</typeparam>
+        /// <returns>An instance of the <typeparamref name="T"/> : <see cref="ConfigFile"/> with values loaded
+        /// from the config file on disk.</returns>
+        T IOptionsPanelHandler.RegisterModOptions<T>()
+        {
+            var configModOptions = new ConfigModOptions<T>();
+            RegisterModOptions(configModOptions);
+
+            var menuAttribute = typeof(T).GetCustomAttributes(typeof(MenuAttribute), true).SingleOrDefault() as MenuAttribute
+                ?? new MenuAttribute();
+
+            if (menuAttribute.SaveOn.HasFlag(MenuAttribute.SaveEvents.SaveGame))
+                IngameMenuHandler.RegisterOnSaveEvent(() => configModOptions.Config.Save());
+            if (menuAttribute.SaveOn.HasFlag(MenuAttribute.SaveEvents.QuitGame))
```

Comment:
Keep an empty line before new if-statements, to more clearly separate them from the previous execution statement.

Reply:
https://github.com/SubnauticaModding/SMLHelper/pull/173/commits/aeda1362ca368e4531a7cfd93cb76712353e8d99